### ISSUE TITLE
Should we allow dot product of sparse vectors?

### ISSUE THREAD

**Issue by shyamupa** [IllinoisCogComp/illinois-sl#10]
Some student trying to use our package said why cannot he take dot product between two sparse vectors. Do we want to provide this? @KaiWeiChang

**Comment by KaiWeiChang** [comment]
Yes. We can provide it, although we do not need this functionality in our current learning algorithms. But it may be useful for other purposes. Could you implement it (it can be modified from the diff function).
